staging/src/k8s.io/component-base/logs/kube-log-runner/kube-log-runner.go

```diff
@@ -26,23 +26,32 @@ import (
 	"os/signal"
 	"strings"
 	"syscall"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/component-base/logs/logrotation"
 )
 
+type OpenFunc func(filePath string, enableFlush bool, maxSize int64, maxAge time.Duration) (io.WriteCloser, error)
+
 var (
+	enableFlush    = flag.Bool("enable-flush", false, "enable flush to log file in every 5 seconds")
 	logFilePath    = flag.String("log-file", "", "If non-empty, save stdout to this file")
+	logFileSize    = flag.String("log-file-size", "0", "Useful with log-file, in format of resource.quantity , if non-zero, rotate log file when it reaches this size")
```

Comment:
```suggestion
        var logFileSize resource.QuantityValue
	flag.Var(&logFileSize "log-file-size", "0", "Useful with log-file, in format of resource.quantity , if non-zero, rotate log file when it reaches this size")
```

Untested, but something roughly like this should work. Then you don't need your custom conversion code.

Reply:
thanks, looks to be working, updated